midtools/plotting.py

```diff
@@ -120,37 +137,45 @@ def plotAllReferencesSAM(
     @param sampleName: The C{str} name of the sample whose reads are being
         analysed.
     """
+    if hbv and not haveHBV:
+        raise ValueError(
+            "You passed a true value for 'hbv' but the pyhbv module could not be "
+            "imported."
+        )
+
     referenceLengths = samFilter.referenceLengths()
     maxReferenceLength = max(referenceLengths.values())
 
     if hbv:
-        referenceScaleFactor = dict(
-            (id_, maxReferenceLength / length)
+        referenceScaleFactor = {
+            id_: maxReferenceLength / length
             for id_, length in referenceLengths.items()
-        )
-        referenceGenotype = dict(
-            (id_, getGenotype(id_) or "UNKNOWN") for id_ in referenceLengths
-        )
+        }
+        referenceGenotype = {
+            id_: (getGenotype(id_) or "UNKNOWN") for id_ in referenceLengths
+        }
     else:
         referenceScaleFactor = dict.fromkeys(referenceLengths, 1.0)
-        referenceGenotype = dict((id_, id_) for id_ in referenceLengths)
+        referenceGenotype = {id_: id_ for id_ in referenceLengths}
 
     genotypes = sorted(set(referenceGenotype.values()), key=genotypeKey)
-    legendRank = dict((genotype, i) for i, genotype in enumerate(genotypes))
+    legendRank = {genotype: i for i, genotype in enumerate(genotypes)}
     nGenotypes = len(genotypes)
     genotypeReferences = defaultdict(set)
 
     data = []
     inLegend = set()
     genotypeCount: Counter[str] = Counter()
 
-    count = 0
-    for count, alignment in enumerate(samFilter.alignments(), start=1):
+    readCount = 0
+    for readCount, alignment in enumerate(samFilter.alignments(), start=1):
         referenceId = alignment.reference_name
+        assert referenceId
         scaleFactor = referenceScaleFactor[referenceId]
         start = alignment.reference_start
+        assert isinstance(alignment.reference_length, int)
         end = start + alignment.reference_length
-        score = alignment.get_tag("AS") + (
+        score = float(alignment.get_tag("AS")) + (
```

Comment:
Why is this needed? Would `int`s be cast to float on the addition anyway and everything that fails the conversion to float also fail the addition?